docs/rfcs/0005-migrate-sitepermission-ac-store-geckoview-store.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: page
+title: Migrate feature-sitepermissions to be compatible with the new GeckoView permission API.
+permalink: /rfc/0005-migrate-sitepermission-ac-store-geckoview-store
+---
+
+* Start date: 2020-07-26
+* RFC PR: [#7855](https://github.com/mozilla-mobile/android-components/pull/7855)
+
+## Summary
+
+Migrate our generic [SitePermissionsStorage](https://github.com/mozilla-mobile/android-components/blob/56e6f5ba55734014804b3f3b0af5c32414fa1d16/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsStorage.kt#L33) to be compatible with the new [GeckoView permissions API](https://docs.google.com/document/d/1KUq0gejnFm5erkHNkswm8JsT7nLOmWvs1KEGFz9FWxk/edit#heading=h.ls1dr18v7zrx).
+
+## Motivation
+
+For permissions, we currently are storing on disk all the site permissions on AC both for GeckoView and WebView, but that's about to change for GeckoView, [a new API is about to land](https://docs.google.com/document/d/1KUq0gejnFm5erkHNkswm8JsT7nLOmWvs1KEGFz9FWxk/edit#heading=h.ls1dr18v7zrx), where GeckoView is going to store all the permissions and we are going to have APIs for managing permissions(query, add, delete, and update).
```

Comment:
nit: We are currently storing all site permissions on disk in AC (both GeckoView and WebView) . With [the new GV API](https://docs.google.com/document/d/1KUq0gejnFm5erkHNkswm8JsT7nLOmWvs1KEGFz9FWxk/edit#heading=h.ls1dr18v7zrx), GeckoView is now going to store all the permissions on their end. This means we need to have APIs for managing permissions(query, add, delete, and update) instead of storing it ourselves.